draft-nygren-httpbis-httpssvc.md

```diff
@@ -574,6 +517,41 @@ Authoritative DNS servers SHOULD return A, AAAA, and HTTPSSVC records
 any in-bailiwick SvcDomainNames.
 
 
+# Performance optimizations
+
+For optimal performance (i.e. minimum connection setup time), clients
+SHOULD issue address (AAAA and/or A) and HTTPSSVC queries
+simultaneously, and SHOULD implement a client-side DNS cache.
+With these optimizations in place, and conforming DNS servers,
+using HTTPSSVC does not add network latency to connection setup.
+
+Using HTTPSSVC with HTTP/3 saves a roundtrip during connection setup,
+compared to HTTP over TLS 1.3 over TCP.  Performance-sensitive
+server operators SHOULD offer HTTP/3 in their HTTPSSVC records.
+
+A nonconforming recursive resolver might return an HTTPSSVC response with
+a nonempty SvcDomainName, without the corresponding address queries.  If
+all the HTTPSSVC RRs in the response have nonempty SvcDomainName values,
+and the client does not have address records for any of these values in
+its DNS cache, the client SHOULD perform an additional address query for
+the SvcDomainName.
+
+The additional DNS query introduces a delay that can be longer or shorter
+than one round trip to the origin.  For simplicity, we assume that these
+two delays will be roughly similar on average, i.e. a 1 RTT delay.
+
```

Comment:
DNS RTTs and Origin RTTs often have no relation and may be orders of magnitude different.  (The former may have cache miss impact and server processing time overhead, but there are also plenty of cases where the DNS recursive is close to the client and the origin is far away, or vice-versa.)

Reply:
Yeah, they're not well-correlated.  The main point here is to find a way to make the average net performance loss negligible (or even a net win), even on legacy recursives (which will surely be prevalent for many years).  My sense is that an acceptable solution:

1. Can't impose a latency-proportional average-case delay on any large category of connections.
2. Must provide security features reliably or not at all.

I'm certainly open to alternative solutions if you don't think this one will be convincing.  I just don't have any other good ideas for how to square this.